disnake/ext/commands/params.py

```diff
@@ -155,7 +147,47 @@ def _xt_to_xe(xe: Optional[float], xt: Optional[float], direction: float = 1) ->
         return None
 
 
-class Injection(Generic[P, T_]):
+def _int_to_str_len(number: int) -> int:
+    """Returns `len(str(number))`, i.e. character count of base 10 signed repr of `number`."""
+    # Desmos equivalent: floor(log(max(abs(x), 1))) + 1 + max(-sign(x), 0)
+    return (
+        int(math.log10(abs(number) or 1))
+        # 0 -> 0, 1 -> 0, 9 -> 0, 10 -> 1
+        + 1
+        + (number < 0)
+    )
+
+
+def _range_to_str_len(min_value: int, max_value: int) -> tuple[int, int]:
+    min_ = _int_to_str_len(min_value)
+    max_ = _int_to_str_len(max_value)
+    opposite_sign = (min_value < 0) ^ (max_value < 0)
+    # both bounds positive: len(str(min_value)) <= len(str(max_value))
+    # smaller bound negative: the range includes 0, which sets the minimum length to 1
+    # both bounds negative: len(str(min_value)) >= len(str(max_value))
+    if opposite_sign:
+        return 1, max(min_, max_)
+    return min(min_, max_), max(min_, max_)
```

Comment:
I haven't looked into it, but assuming we can expect `min_value < max_value` to always be true here, I believe this could be simplified a little